### PR TITLE
require MODULE_NAME

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

**Comment by dagar** [comment]
This gets rid of the Unknown prints and also helps to identify low level dependencies that shouldn't be using orb.

File: src/drivers/samv7/drv_hrt.c [hunk 1]
```diff
@@ -347,7 +347,7 @@ hrt_tim_init(void)
 	ret = sam_tc_clockselect(frequency, &cmr, &actual);
 
 	if (ret < 0) {
-		PX4_PANIC("ERROR: Failed no divisor can be found (%d),for timer channel %d\n", ret, HRT_TIMER_CHANNEL);
+		printf("ERROR: Failed no divisor can be found (%d),for timer channel %d\n", ret, HRT_TIMER_CHANNEL);
```
Comment (davids5) [inline src/drivers/samv7/drv_hrt.c]: Why doesn't PX4_PANIC do the correct platform specific thing?

**Comment by dagar** [comment]
PX4_PANIC doesn't appear to be anything other than a level (no assert).

**Comment by davids5** [comment]
@dagar not good

**Comment by dagar** [comment]
PX4_PANIC = PX4 error + NuttX PANIC()?

**Comment by davids5** [comment]
@dagar - lets define PANIC() for all platforms. Then yes it would always be PX4 error + PANIC()

PANIC on all platforms should print short filename, line number and exit() (hardfault)

This is one of those situations where "Nebbles Law applies" - "It is not how it works, when it works, it is how it works, when it doesn't work that counts"

The thing to watchout for is any complex logic or os functions and data in the path. I worry about logger etal in the path.

In the Nuttx world it is a syslog write. (There are data structures and objects in memory that are subject to corruption) In the old days it was up_loputc: A per character wait on USART TX ready - write byte to USART DR. The has very little probability of memory corruption that would prevent the output from reaching the user's terminal but you will not get it "Logged" unless the hard fault is logged on the next boot.

So we should consider adding a protected store and try to do the same thing as in hardfault_log on CPU sans BBSRAM.

**Comment by dagar** [comment]
@davids5 create a new issue for PX4_PANIC requirements and merge this?

**Comment by davids5** [comment]
yes

**Pull request merged**